e2e/Cargo.toml

```diff
@@ -11,4 +11,4 @@ portpicker = "0.1.1"
 rand = "0.8.5"
 reqwest = { version = "0.11.10", features = ["blocking", "json"] }
 which = "4.2.5"
-tempdir = "0.3.7"
+tempfile = "3.3.0"
```

Comment:
Since there are no other updates for `e2e`, does that mean `e2e` does not use `tempdir`?

Reply:
apologies did that with fuzzy search, didn't check that

Reply:
Okay, if we can remove this line then lets remove it

Reply:
We do use tempdir in the e2e tests @gautamprikshit1, specifically in `tests/integration/helpers`, but it might not be used when you run locally. So we should also update tempdir to tempfile here. By the way, if you haven't tried it yet, there are instructions for how to run the e2e tests in the e2e directory README.